scripts/browse.py

```diff
@@ -80,36 +82,72 @@ def create_message(chunk, question):
         "content": f"\"\"\"{chunk}\"\"\" Using the above text, please answer the following question: \"{question}\" -- if the question cannot be answered using the text, please summarize the text."
     }
 
-def summarize_text(text, question):
-    if not text:
-        return "Error: No text to summarize"
 
-    text_length = len(text)
-    print(f"Text length: {text_length} characters")
+@retry(wait=wait_exponential(multiplier=1, min=4, max=30), stop=stop_after_attempt(5))
+def call_create_chat_completion(messages):
```

Comment:
I would suggest moving this into the create_chat_completion file and naming it something like create_chat_completion_with_retry.

Reply:
done